pagegraph/src/graph_algos.rs

```diff
@@ -798,6 +798,42 @@ impl PageGraph {
 
         already_checked
     }
+
+    pub fn all_downstream_requests_nested<'a>(&'a self, edge: &'a Edge) -> Vec<DownstreamRequests> {
```

Comment:
nit: I think the `_nested` suffix was meant to distinguish this from a top-level caller? It can probably be dropped since we're doing it all in one function now.

Reply:
I thought it might be worth leaving the old one around - this file is a library of functions, so we could distinguish between returning the the flat list of downstream requests vs nested/tree one.

Reply:
Ah gotcha, I just misunderstood the intent of the suffix. Looks good then!